### DISCUSSION TITLE
Better window chrome on Wayland Gnome?

### DISCUSSION THREAD

**Discussion by HubKing** [tsujan/FeatherPad#709] [Q&A] [answered]
To be fair, this is NOT FeatherPad's unique problem, but it seems a problem of QT6 apps on Wayland Gnome, because another QT6 app Strawberry has the exact same problem.

The title bar is a lot thinner than GTK3 apps', and it does not work to double-click maximise the window, because double clicking is somehow interpreted as small dragging. My guess is that it miscalculates the coordinate (maybe because of fractional scaling?).

I wonder if there is any way to 

* make the title bar taller and buttons bigger
* double-click to maximise

-----

Comparison with  a GTK3 app's chrome (LibreOffice). QT6's title bar is a lot thinner, with small buttons.

![image](https://user-images.githubusercontent.com/10900489/225624108-eb329826-ae86-4883-a2bc-2bd537067e09.png)

Double clicking

https://user-images.githubusercontent.com/10900489/225623219-a9a9c7c8-98ae-4daf-ba92-6b1b5d897787.mp4

**Comment by tsujan** [comment, marked as answer]
The tittle-bar is a part of the decoration. There are two kinds of decoration: server-side (SSD) and client-side (CSD). Wayland DEs may give Qt apps an SSD (like in Plasma-Wayland or, optionally, Wayfire) or leave them with Qt's poor CSD (optional in Wayfire). I don't know what Gnome does, but these things are outside apps' control.


SSD can always be customized in DE's settings dialogs; Qt's CSD can't. At some point, Qt's CSD might be customizable by using `QStyle`, but that will also outside apps' control (`QStyle` is controlled by Qt style plugins, like Fusion, Breeze, Kvantum,...).

I don't have Gnome but, as far as I remember, it gave Qt5 apps an SSD. Maybe, it can't handle Qt6 apps yet.

**Reply by HubKing** [reply]
Yeah, I know that the chrome was done by some sort of QT library outside of the app, but I was wondering if QT6 apps could sort-of customise them by passing some sort of parameters when creating the window. If there is nothing the application developers do other than re-compiling that QT's window chrome library, which should be too much work, then I guess I have to give it up.

I have tried using QT6CT, but it had no option for window chrome.

If I remember correctly, when QT6 apps are run on Wayland KDE, they get the system's native chrome (i.e., the things that the user has set in KDE's settings), so I think this problem only exists in Gnome.  QT developers don't like Gnome or something? I don't understand why window chrome is so awkward on Wayland Gnome.

**Reply by tsujan** [reply]
>  but I was wondering if QT6 apps could sort-of customise them...

I explained it above.

BTW, the word "chrome" isn't used here.

>  QT developers don't like Gnome or something?

As far as I've seen, it's rather Gnome that doesn't like anything other than Gnome. However, as I said above, Qt's CSD is poor for now, regardless of Gnome — you can say, Qt devs may not like CSD — but that might change at some point.

**Reply by HubKing** [reply]
It seems that I had a wrong idea about chrome. I thought it was the so-called "non-client area" of GUI windows, but it seems that it means the GUI itself.

Anyway. I hope QT developers would fix this, because this makes QT6 apps annoying to use on Gnome.

**Comment by tsujan** [comment]
> I hope QT developers would fix this, because this makes QT6 apps annoying to use on Gnome.

As the author of a Qt style plugin (Kvantum), I hoped that Qt6 would add `QStyle` parameters for its CSD, so that it could be customized by using the plugin's config dialog; but that didn't happen.

Wayland compositors like Wayfire solve such problems by having an option for SSD. KWin solves it by only using SSD.
